crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -16,413 +21,426 @@ use {
-        },
-    ];
-    // Without protocol fee:
-    // Expected execution is 10000000000000000000 GNO for
-    // 9871415430342266811 DAI, with executed_surplus_fee = 167058994203399 GNO
-    //
-    // With protocol fee:
-    // Expected executed_surplus_fee is 167058994203399 +
-    // 0.1*(10000000000000000000 - 167058994203399) = 1000150353094783059
-    //
-    // Final execution is 10000000000000000000 GNO for 8884273887308040129 DAI, with
-    // executed_surplus_fee = 1000150353094783059 GNO
-    //
-    // Settlement contract balance after execution = 1000150353094783059 GNO =
-    // 1000150353094783059 GNO * 8884273887308040129 / (10000000000000000000 -
-    // 1000150353094783059) = 987306456662572858 DAI
-    execute_test(
-        web3.clone(),
-        vec![ProtocolFeesConfig(protocol_fees)],
+    .await
+    .unwrap();
+    let market_quote_before = get_quote(
+        &services,
+        onchain.contracts().weth.address(),
+        market_order_token.address(),
         OrderKind::Sell,
-        None,
-        1000150353094783059u128.into(),
-        987306456662572858u128.into(),
+        sell_amount,
+        quote_valid_to,
     )
-    .await;
-}
-
-async fn volume_fee_sell_order_test(web3: Web3) {
-    let fee_policy = FeePolicyKind::Volume { factor: 0.1 };
-    // Tests ability of providing `any` fee policy order class config. In fact, the
-    // Market config could be used since the order is in-market.
-    let protocol_fee = ProtocolFee {
-        policy: fee_policy,
-        policy_order_class: FeePolicyOrderClass::Any,
-    };
-    // Without protocol fee:
-    // Expected execution is 10000000000000000000 GNO for
-    // 9871415430342266811 DAI, with executed_surplus_fee = 167058994203399 GNO
-    //
-    // With protocol fee:
-    // Expected executed_surplus_fee is 167058994203399 +
-    // 0.1*(10000000000000000000 - 167058994203399) = 1000150353094783059
-    //
-    // Final execution is 10000000000000000000 GNO for 8884273887308040129 DAI, with
-    // executed_surplus_fee = 1000150353094783059 GNO
-    //
-    // Settlement contract balance after execution = 1000150353094783059 GNO =
-    // 1000150353094783059 GNO * 8884273887308040129 / (10000000000000000000 -
-    // 1000150353094783059) = 987306456662572858 DAI
-    execute_test(
-        web3.clone(),
-        vec![ProtocolFeesConfig(vec![protocol_fee])],
+    .await
+    .unwrap();
+    let partner_fee_quote_before = get_quote(
+        &services,
+        onchain.contracts().weth.address(),
+        partner_fee_order_token.address(),
         OrderKind::Sell,
-        None,
-        1000150353094783059u128.into(),
-        987306456662572858u128.into(),
+        sell_amount,
+        quote_valid_to,
     )
-    .await;
-}
+    .await
+    .unwrap();
+
+    let market_price_improvement_order = OrderCreation {
+        sell_token: onchain.contracts().weth.address(),
+        sell_amount,
+        buy_token: market_order_token.address(),
```

Comment:
nit: Since we already have the quote it seems more error resistant to re-use the values from the quote whenever possible.
I.e. `buy_token: market_quote_before.quote.buy_token` or maybe even better sth like:
```rust
OrderCreation {
    buy_amount: market_quote_before.quote.buy_amount * 2 / 3,
    ..order_from_quote(market_quote_before)
}
```
That way it's absolutely clear that this order was created from a quote and explicitly overwrites a single value to make it work for the test.

Also since we have fixed size arrays we can probably do sth like:
```rust
let [market_quote, limit_quote, _parter_fee_quote] = [market_token, limit_token, partner_fee_token].map(|token| /*fetch quote*/);
```

Reply:
Refactored